src/network/connection.js

```diff
@@ -271,6 +271,11 @@ module.exports = class Connection {
     await this.requestQueue.waitForPendingRequests()
     this.requestQueue.destroy()
 
+    this.bytesNeeded = Decoder.int32Size()
```

Comment:
Can we just add a quick comment explaining why we are doing this here?

Reply:
Do you mean around #274 in particular?

Reply:
These 4 lines that were added.